src/mesh/Mesh.cpp

```diff
@@ -146,6 +147,9 @@ PtrData &Mesh::createData(
   //#rows = dimensions of current mesh #columns = dimensions of corresponding data set
   PtrData data(new Data(name, id, dimension, _dimensions));
   _data.push_back(data);
+  // @todo strange, but would otherwise require a lot of refactoring of Waveform class.
+  data->initializeWaveform(data, waveformDegree);
```

Comment:
yes, a bit strange indeed, but maybe also a sign that sth could be simpler?

Reply:
I'm done moving things around. Now `Waveform` owns `time::Storage` and this simplifies a lot of things.